src/main/java/Duke.java

```diff
@@ -1,10 +1,174 @@
+            Label dukeText = new Label(dukeRun(userInput.getText()));
+            dialogContainer.getChildren().addAll(
+                    DialogBox.getUserDialog(userText, new ImageView(dukeUser)),
+                    DialogBox.getDukeDialog(dukeText, new ImageView(dukeServer))
+            );
+            userInput.clear();
+        });
+
+        userInput.setOnAction((event) -> {
+            Label userText = new Label(userInput.getText());
+            Label dukeText = new Label(dukeRun(userInput.getText()));
+            dialogContainer.getChildren().addAll(
+                    DialogBox.getUserDialog(userText, new ImageView(dukeUser)),
+                    DialogBox.getDukeDialog(dukeText, new ImageView(dukeServer))
+            );
+            userInput.clear();
+        });
+
+    }
+
+    /**
+     * Processing method for GUI version of Duke.
+     * @param input String value from user typing in textfield
+     * @return String output for Duke to reply
+     */
+    public String dukeRun(String input) {
+        String reply = "";
+        if (input.toLowerCase().equals("hello")) {
+            reply = user.welcome() + user.greeting();
+            assert !reply.isEmpty() : "Reply should not be empty";
+        } else if (input.toLowerCase().equals("bye")) {
+            //store current tasks into local file for future reference
+            store.storeCurrent(parse.getList());
+            Platform.exit();
+        } else if (input.toLowerCase().contains("done")) {
+            try {
+                //mark task as done, change cross to tick
+                int itemNum = Integer.parseInt(input.substring(input.length() - 1));
+                Task curr = parse.getList().get(itemNum - 1);
+                curr.markAsDone();
+                assert curr.isDone : "Task should have been marked as done";
+                //forming the message
+                reply = user.doneMessage(curr);
+            } catch (IndexOutOfBoundsException e) {
+                reply = user.indexError();
+            } catch (NumberFormatException e) {
+                reply = user.emptyError();
+            }
+        } else if (input.toLowerCase().equals("list")) {
+            reply = user.listTask(parse.getList());
+        } else if (input.toLowerCase().contains("delete")) {
+            try {
+                int itemNum = Integer.parseInt(input.substring(input.length() - 1));
+                reply = parse.deleteTask(itemNum);
+            } catch (IndexOutOfBoundsException e) {
+                reply = user.indexError();
+            } catch (NumberFormatException e) {
+                reply = user.emptyError();
+            }
+        } else if (input.toLowerCase().contains("find")) {
+            String searchTerm = input.substring(5);
+            reply = user.findTask(parse.getList(), searchTerm);
+        } else {
+            reply = parse.generateTask(input);
+        }
+
+        return reply;
```

Comment:
Perhaps you might want to consider applying more layers of abstraction, like putting this block in an Ui class or Parser class. SLAP dat.